smartsim/_core/generation/generator.py

```diff
@@ -59,12 +65,28 @@ def __init__(self, gen_path: str, overwrite: bool = False) -> None:
         is false, raises EntityExistsError when there is a name
         collision between entities.
 
+        :param gen_path: Path in which files need to be generated
+        :type gen_path: str
         :param overwrite: toggle entity replacement, defaults to False
         :type overwrite: bool, optional
+        :param verbose: Whether generation information should be logged to std out
+        :type verbose: bool, optional
         """
         self._writer = ModelWriter()
         self.gen_path = gen_path
         self.overwrite = overwrite
+        self.log_level = DEBUG if not verbose else INFO
+
+    @property
+    def log_file(self) -> str:
+        """Returns the location of the file
+        summarizing the parameters used for the last generation
+        of all generated entities.
+
+        :returns: path to file with parameter settings
+        :rtype: str
+        """
+        return join(self.gen_path, "smartsim_params.txt")
```

Comment:
should we shift default `smartsim_params.txt` into `smartsim._core.config.config` and retrieve it from there instead?